wgpu-core/Cargo.toml

```diff
@@ -120,36 +120,47 @@ static-dxc = ["wgpu-hal/static-dxc"]
 # platform crates collude to provide platform-specific behavior.
 
 ## DX12 backend
-dx12 = ["wgpu-core-platform-windows-linux-android-emscripten/dx12"]
+dx12 = ["wgpu-core-deps-windows-linux-android/dx12"]
 ## Metal backend
-metal = ["wgpu-core-platform-apple/metal"]
+metal = ["wgpu-core-deps-apple/metal"]
 ## Vulkan backend, only available on Windows, Linux, Android
-vulkan = ["wgpu-core-platform-windows-linux-android-emscripten/vulkan"]
+vulkan = ["wgpu-core-deps-windows-linux-android/vulkan"]
 ## OpenGL backend, only available on Windows, Linux, Android, and Emscripten
-gles = ["wgpu-core-platform-windows-linux-android-emscripten/gles"]
+gles = [
+    "wgpu-core-deps-windows-linux-android/gles",
+    "wgpu-core-deps-emscripten/gles",
+]
 
 ## WebGL backend, only available on Emscripten
-webgl = ["wgpu-core-platform-wasm/webgl"]
+webgl = ["wgpu-core-deps-wasm/webgl"]
 ## OpenGL backend, on macOS only
-angle = ["wgpu-core-platform-apple/angle"]
+angle = ["wgpu-core-deps-apple/angle"]
 ## Vulkan portability backend, only available on macOS
-vulkan-portability = ["wgpu-core-platform-apple/vulkan-portability"]
+vulkan-portability = ["wgpu-core-deps-apple/vulkan-portability"]
 ## Renderdoc integration, only available on Windows, Linux, and Android
-renderdoc = ["wgpu-core-platform-windows-linux-android-emscripten/renderdoc"]
+renderdoc = ["wgpu-core-deps-windows-linux-android/renderdoc"]
 
 ## Enable the `noop` backend.
 # TODO(https://github.com/gfx-rs/wgpu/issues/7120): there should be a hal feature
 noop = []
 
+# The target limitation here isn't needed, but prevents more than one of these
+# platform crates from being included in the build at a time, preventing users
+# from getting confused by seeing them in the list of crates.
+[target.'cfg(target_vendor = "apple")'.dependencies]
+wgpu-core-deps-apple = { workspace = true, optional = true }
+[target.'cfg(target_os = "emscripten")'.dependencies]
+wgpu-core-deps-emscripten = { workspace = true, optional = true }
+[target.'cfg(all(target_arch = "wasm32", not(target_os = "emscripten")))'.dependencies]
+wgpu-core-deps-wasm = { workspace = true, optional = true }
+[target.'cfg(any(windows, target_os = "linux", target_os = "android"))'.dependencies]
+wgpu-core-deps-windows-linux-android = { workspace = true, optional = true }
```

Comment:
clever! it's technically also a lil bit faster because then there's a smaller number of crates that cargo has to dispatch and makes the  `--timings` report better